src/USER-MISC/angle_fourier_simple_approx.cpp

```diff
@@ -0,0 +1,218 @@
+ * Absolute error <= 6.7e-5
+ * Range mapping not necessary as C++ standard library would throw domain error
+ * Source: https://developer.download.nvidia.com/cg/acos.html
+ */
+double AngleFourierSimpleApprox::fastAcos(double x) {
+  double negate = double(x < 0);
+  x = abs(x);
+  double ret = -0.0187293;
+  ret = ret * x;
+  // TODO: verify that fpas are used here
+  ret = ret + 0.0742610;
+  ret = ret * x;
+  ret = ret - 0.2121144;
+  ret = ret * x;
+  ret = ret + 1.5707288;
+  ret = ret * sqrt(1.0-x); 
+  ret = ret - 2 * negate * ret;
+  return negate * 3.14159265358979 + ret;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void AngleFourierSimpleApprox::compute(int eflag, int vflag)
+{
+  ev_init(eflag,vflag);
+
+  const int nall = atom->nlocal + atom->nghost;
+  const int inum = neighbor->nanglelist;
+
+      if (evflag) {
+        if (eflag) {
+          if (force->newton_bond) eval<1,1,1>();
+          else eval<1,1,0>();
+        } else {
+          if (force->newton_bond) eval<1,0,1>();
+          else eval<1,0,0>();
+        }
+      } else {
+        if (force->newton_bond) eval<0,0,1>();
+        else eval<0,0,0>();
+      }
+}
+
+template <int EVFLAG, int EFLAG, int NEWTON_BOND>
+void AngleFourierSimpleApprox::eval()
+{
+  int i1,i2,i3,n,type;
+  double delx1,dely1,delz1,delx2,dely2,delz2;
+  double eangle,f1[3],f3[3];
+  double term,sgn;
+  double rsq1,rsq2,r1,r2,c,cn,th,nth,a,a11,a12,a22;
+  int nanglelist = neighbor->nanglelist;
+
+  const dbl3_t * _noalias const x = (dbl3_t *) atom->x[0];
+  dbl3_t * _noalias const f = (dbl3_t *) atom->f[0];
+  const int4_t * _noalias const anglelist = (int4_t *) neighbor->anglelist[0];
+  const int nlocal = atom->nlocal;
+  eangle = 0.0;
+
+  for (n = 0; n < nanglelist; n++) {
+    i1 = anglelist[n].a;
+    i2 = anglelist[n].b;
+    i3 = anglelist[n].c;
+    type = anglelist[n].t;
+
+    // 1st bond
+
+    delx1 = x[i1].x - x[i2].x;
+    dely1 = x[i1].y - x[i2].y;
+    delz1 = x[i1].z - x[i2].z;
+
+    rsq1 = delx1*delx1 + dely1*dely1 + delz1*delz1;
+    r1 = sqrt(rsq1);
+
+    // 2nd bond
+
+    delx2 = x[i3].x - x[i2].x;
+    dely2 = x[i3].y - x[i2].y;
+    delz2 = x[i3].z - x[i2].z;
+
+    rsq2 = delx2*delx2 + dely2*dely2 + delz2*delz2;
+    r2 = sqrt(rsq2);
+
+    // angle (cos and sin)
+
+    c = delx1*delx2 + dely1*dely2 + delz1*delz2;
+    c /= r1*r2;
+
+    if (c > 1.0) c = 1.0;
+    if (c < -1.0) c = -1.0;
+
+    // force & energy
+
+    th = fastAcos(c);
+    nth = N[type]*fastAcos(c);
+    cn = fastCos(nth);
+    term = k[type]*(1.0+C[type]*cn);
+
+    if (EFLAG) eangle = term;
+
+    // handle sin(n th)/sin(th) singulatiries
+
+    if (fabs(c)-1.0 > 0.0001) {
+      a = k[type]*C[type]*N[type]*sin(nth)/sin(th);
+    } else {
+      if (c >= 0.0) {
+        term = 1.0 - c;
+        sgn = 1.0;
+      } else {
+        term = 1.0 + c;
+        sgn = ( fmodf((float)(N[type]),2.0) == 0.0f )?-1.0:1.0;
```

Comment:
This should be double precision. I'll include the corresponding change in the upstream non-approximated version
(cf. https://github.com/lammps/lammps/commit/455b7b93929040e977ecf35ee39d7350e25b7a66 )